src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -211,6 +218,34 @@ func (mod *mgmtModule) getFabricInterface(ctx context.Context, numaNode int, net
 	return mod.fabricInfo.GetDevice(numaNode, netDevClass, provider)
 }
 
+func (mod *mgmtModule) waitFabricReady(ctx context.Context, netDevClass hardware.NetDevClass) error {
+	if mod.netIfaces == nil {
+		mod.netIfaces = net.Interfaces
+	}
```

Comment:
Is there a potential race here? What happens if two different `HandleCall()` invocations come in nearly simultaneously?

Reply:
Good point. Will fix.

Reply:
Actually, thinking about this, there is a general race if a lot of requests come in at once. I'm going to push a possible solution with a mutex -- this could really damage performance if someone isn't using the cache though.